test/gurka/test_route_summary.cc

```diff
@@ -131,3 +121,98 @@ TEST(TestRouteSummary, GetSummary) {
+                  /                     \              / \
+      A----------B                       E------------F   J
+                 |                       |             \ /
+                 |                       |              K
+                 |                       |
+                 |                       |
+                 |                       |
+                 C-----------------------D
+    )";
+
+  const gurka::ways ways = {
+      {"AB", {{"highway", "primary"}, {"name", "RT 1"}}},
+      {"BGHE", {{"highway", "motorway"}, {"name", "RT 1"}}},
+      {"EF", {{"highway", "primary"}, {"name", "RT 2"}}},
+      {"FIJ", {{"highway", "motorway"}, {"name", "RT 3"}}},
+      {"BCDE", {{"highway", "primary"}, {"name", "RT 2"}}},
+      {"FKJ", {{"highway", "primary"}, {"name", "RT 4"}, {"access", "no"}, {"foot", "yes"}}},
+  };
+
+  const auto node_layout = gurka::detail::map_to_coordinates(ascii_map, 100, PointLL{5.108, 52.01});
+
+  std::unordered_map<std::string, std::string> config_map =
+      {{"mjolnir.data_processing.use_direction_on_ways", "true"},
+       {"mjolnir.admin", VALHALLA_SOURCE_DIR "test/data/netherlands_admin.sqlite"}};
+
+  std::string workdir = "test/data/gurka_test_dup_summary_fix";
+  valhalla::gurka::map map = gurka::buildtiles(node_layout, ways, {}, {}, workdir, config_map);
+
+  map.nodes = node_layout;
+
+  if (!filesystem::exists(workdir)) {
+    bool created = filesystem::create_directories(workdir);
+    EXPECT_TRUE(created);
+  }
+
+  // auto will take the short route via the motorway
+  valhalla::Api result0 = gurka::do_action(valhalla::Options::route, map, {"A", "J"}, "auto");
+  EXPECT_EQ(result0.trip().routes_size(), 1);
+  EXPECT_EQ(result0.trip().routes(0).legs_size(), 1);
+  gurka::assert::osrm::expect_steps(result0, {"RT 1", "RT 2", "RT 3"});
+  gurka::assert::osrm::expect_summaries(result0, {"RT 1, RT 2"});
+
+  // Pedestrians avoid the motorways
+  valhalla::Api result1 = gurka::do_action(valhalla::Options::route, map, {"A", "J"}, "pedestrian");
+  EXPECT_EQ(result1.trip().routes_size(), 1);
+  EXPECT_EQ(result1.trip().routes(0).legs_size(), 1);
+  gurka::assert::osrm::expect_steps(result1, {"RT 1", "RT 2", "RT 4"});
+  gurka::assert::osrm::expect_summaries(result1, {"RT 1, RT 2"});
+
+  // You can see above that all two results have the same summary: RT 1, RT 2. However, each
+  // chose different paths. The auto's longest named segment was RT 1 and the ped's longest
+  // named segment was RT 2. Because the longest legs were different, the summary logic
+  // stopped searching after the first named segment and returned the minimum number of
+  // named segments (2) which resulted in them both having the same summary: RT 1, RT 2.
+  // The route's can be made unique by having the auto choose RT 3 and the ped choose RT 4
+  // at the end of the route.
+
+  // Below we mash together the three above results into a single result. When we
```

Comment:
looking below it seems like actually you wanted to have alternates that you needed to generate unique summaries for? you can get alternates by asking for it in the request `"alternates":1`. maybe the map above fails to find an alternate? anyway i think it would simplify and make the test a bit more understandable.

originally when i read the comment above you happened to use the word `leg` which made me think of multipoint route but that isnt what you meant, you really meant just a subsection (series of edges) of the route.

or have i confused what you meant?

Reply:
Hmm, that's a good suggestion about `"alternates":1`.  However, wrestling gurka into the configuration I desire can sometimes take more time than expected.  So if its alright I'd like to leave it as it is:  not really testing whether we _can_ generate alternates and produce unique summaries - but rather, just a test of unique summary generation.

And sorry, yes, I frequently catch myself typing comments with the term `leg` to just mean "some section of a route", realizing it is an important reserved valhalla term. I'll keep working on that. 😅

Reply:
Oh i agree we arent testing whether or not we can generate alternates we are testing whether or not alternates have unique names. I just think that your test can be simplified to something like:

```c++
  valhalla::Api result0 = gurka::do_action(valhalla::Options::route, map, {"A", "J"}, "auto", {{"alternates", "1"}});
  EXPECT_EQ(result.trip().routes_size(), 2);
  EXPECT_EQ(result.trip().routes(0).legs_size(), 1);
  EXPECT_EQ(result.trip().routes(1).legs_size(), 1);
  const std::string expected_route_summary0 = "RT 1, RT 2, RT 3";
  const std::string expected_route_summary1 = "RT 1, RT 2, RT 4";
  gurka::assert::osrm::expect_summaries(result, {expected_route_summary0, expected_route_summary1});
```

I find that a heck of a lot easier to understand then the the splicing etc

Reply:
I'll give this a quick shot... if it isn't quick I'd like to punt and move on with other things.  Brb.

Reply:
I took a quick shot at modifying the test to actually compute alternates.  Not sure why it didn't, I was still only getting one route.  I was able to remove the pedestrian-only attributes, fwiw.  I guess rather than spend more time simplifying this test (which sufficiently proves the fix), I'd like to move on with other things.  thanks.

Reply:
no worries, thanks for taking a shot